core/capabilities/remote/target/server.go

```
@@ -156,7 +157,7 @@ func (r *server) Receive(ctx context.Context, msg *types.MessageBody) {
 	if len(requestIDs) > 1 {
 		// This is a potential attack vector as well as a situation that will occur if the client is sending non-deterministic payloads
 		// so a warning is logged
-		r.lggr.Warnw("received messages with the same id and different payloads", "messageID", messageId, "requestIDToCount", requestIDs)
+		r.lggr.Warnw("received messages with the same id and different payloads", "messageID", remote.SanitizeLogString(messageId), "lenRequestIDs", len(requestIDs))
```

would it be better to validate the message ID in 'GetMessageID' function? then the majority of the log sanitization would be unnecessary in this file and would prevent any accidental logging of malicious ids returned in errors as well.

Also, the same would apply to client.go as there are places where we log message IDs in that file that are not captured in this PR that probably should be (e.g line 156).   In addition I think similar validation in method 'GetMessageIDForRequest' should be applied to obviate the need  for log sanitisation.

I also think validating at this point is better as it would early fail rather than continue to try to process a malicious message with sanitised logging.  Perhaps I am missing something?

Just a thought.....maybe we should avoid the target and trigger shims from handling the raw messages directly and instead wrap the messages in a wrapper than can enforce validation of things like the ID/method name ( perhaps a custom string()  as well) and have the remote shims only handle the wrapped types?

---

Yeah that's reasonable, let me look into it.